File: python/insert_delete_getrandom_O1.py
# no. 380
import random
class RandomizedSet:

    def __init__(self):
        self.num_set = []
        return

    def insert(self, val: int) -> bool:
        for i in self.num_set:
            if i == val:
                return False
        self.num_set.append(val)
        return True

    def remove(self, val: int) -> bool:
        for i in range(len(self.num_set)):
            if self.num_set[i] == val:
                del self.num_set[i]
                return True
        return False

    def getRandom(self) -> int:
        return random.choice(self.num_set)

File: C++/insert_delete_getrandom_O1.cpp
// no. 380
#include <cstdlib>
#include <vector>
using namespace std;

class RandomizedSet {
public:
  vector<int> num_set;
  RandomizedSet() {}

  bool insert(int val) {
    for (int i : num_set) {
      if (i == val) {
        return false;
      }
    }
    num_set.push_back(val);
    return true;
  }

  bool remove(int val) {
    for (int i = 0; i < num_set.size(); i++) {
      if (num_set[i] == val) {
        num_set.erase(num_set.begin() + i);
        return true;
      }
    }
    return false;
  }

  int getRandom() {
    int min = 0;
    int max = num_set.size();
    int rand_index = rand() % (max - min);
    return num_set[rand_index];
  }
};
